src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/TouchEffect.shared.cs

```diff
@@ -0,0 +1,1145 @@
+			typeof(double),
+			typeof(TouchEffect),
+			0.0,
+			propertyChanged: ForceUpdateStateAndTryGenerateEffect);
+
+		public static readonly BindableProperty HoveredRotationXProperty = BindableProperty.CreateAttached(
+			nameof(HoveredRotationX),
+			typeof(double),
+			typeof(TouchEffect),
+			0.0,
+			propertyChanged: ForceUpdateStateAndTryGenerateEffect);
+
+		public static readonly BindableProperty PressedRotationXProperty = BindableProperty.CreateAttached(
+			nameof(PressedRotationX),
+			typeof(double),
+			typeof(TouchEffect),
+			0.0,
+			propertyChanged: ForceUpdateStateAndTryGenerateEffect);
+
+		public static readonly BindableProperty NormalRotationYProperty = BindableProperty.CreateAttached(
+			nameof(NormalRotationY),
+			typeof(double),
+			typeof(TouchEffect),
+			0.0,
+			propertyChanged: ForceUpdateStateAndTryGenerateEffect);
+
+		public static readonly BindableProperty HoveredRotationYProperty = BindableProperty.CreateAttached(
+			nameof(HoveredRotationY),
+			typeof(double),
+			typeof(TouchEffect),
+			0.0,
+			propertyChanged: ForceUpdateStateAndTryGenerateEffect);
+
+		public static readonly BindableProperty PressedRotationYProperty = BindableProperty.CreateAttached(
+			nameof(PressedRotationY),
+			typeof(double),
+			typeof(TouchEffect),
+			0.0,
+			propertyChanged: ForceUpdateStateAndTryGenerateEffect);
+
+		public static readonly BindableProperty AnimationDurationProperty = BindableProperty.CreateAttached(
+			nameof(AnimationDuration),
+			typeof(int),
+			typeof(TouchEffect),
+			default(int),
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty AnimationEasingProperty = BindableProperty.CreateAttached(
+			nameof(AnimationEasing),
+			typeof(Easing),
+			typeof(TouchEffect),
+			null,
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty PressedAnimationDurationProperty = BindableProperty.CreateAttached(
+			nameof(PressedAnimationDuration),
+			typeof(int),
+			typeof(TouchEffect),
+			default(int),
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty PressedAnimationEasingProperty = BindableProperty.CreateAttached(
+			nameof(PressedAnimationEasing),
+			typeof(Easing),
+			typeof(TouchEffect),
+			null,
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty NormalAnimationDurationProperty = BindableProperty.CreateAttached(
+			nameof(NormalAnimationDuration),
+			typeof(int),
+			typeof(TouchEffect),
+			default(int),
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty NormalAnimationEasingProperty = BindableProperty.CreateAttached(
+			nameof(NormalAnimationEasing),
+			typeof(Easing),
+			typeof(TouchEffect),
+			null,
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty HoveredAnimationDurationProperty = BindableProperty.CreateAttached(
+			nameof(HoveredAnimationDuration),
+			typeof(int),
```

Comment:
Using a `Timespan` together with a TypeConverter perhaps opens up more options.

Reply:
Decided to leave int due to attached properties (discussed over Discord)